src/Http/RequestFactory.php

```diff
@@ -185,24 +187,34 @@ public function createHttpRequest()
 			}
 		}
 
+		$remoteAddr = !empty($_SERVER['REMOTE_ADDR']) ? $_SERVER['REMOTE_ADDR'] : NULL;
+		$remoteHost = !empty($_SERVER['REMOTE_HOST']) ? $_SERVER['REMOTE_HOST'] : NULL;
```

Comment:
Having invalid value like '0' here makes no sense either and it's not checked anyway, maybe it could be.